examples/wallet/app/components/InputKeys.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import routes from '../constants/routes.json';
+import typeof { setAccount as SetAccount } from '../actions/account';
+import typeof { updateBalance as UpdateBalance } from '../actions/balance';
+import getAccountFromPrivateKey from '../utils/wasmWrapper';
+
+type Props = {
+  setAccount: SetAccount,
+  updateBalance: UpdateBalance,
+  privateKey: string
+};
+
+// FIXME: this has no error handling, neither while parsing the address
+// nor when fetching the balance.
+export default ({ updateBalance, setAccount, privateKey }: Props) => {
+  const handleSubmit = function handleSubmit(event) {
+    event.preventDefault();
+    return getAccountFromPrivateKey(newPrivateKey)
+      .then(setAccount)
+      .then(updateBalance);
+  };
+  const [newPrivateKey, setNewPrivateKey] = useState(privateKey);
+  if (privateKey) {
```

Comment:
just nitpicking, wouldn't this redirect  be a bit more clear if done before setting the hook?